### DISCUSSION TITLE
Replace DataGrid with TemplatedDataGrid or DataBox

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#6862] [Ideas]
Hi, 
in the test project I am working on I have numerous problems with the DataGrid. Datagrid is very buggy especially for virtualization.

Perhaps in future major versions of Avalonia it should be replaced with something else.

Personally, I got along very well with these projects:

Both are based on Avalonia ListBox

The pros and cons of a possible future migration:

## Pro
- Very clean and manageable code
- Virtualization works fine, inherits from Avalonia ListBox
- When Avalonia ListBox supports row grouping, both projects will benefit from some changes

## Cons
- Currently both projects do not support row grouping

**Comment by wieslawsoltes** [comment]
Please note that TemplatedDataGrid will be deprecated and all functionality will be moved to DataBox (almost done) as the TemplatedDataGrid has few I think hard to fix issues (layout glitches when vertically resizing and stability issues with grid shared size scope) but those are fixed in DataBox. Also resizing columns is harder to implement in TemplatedDataGrid as it uses grid splitter which is broken in more complex column configurations (same as in WPF).
